### ISSUE TITLE
SELECT with Enum8 as index missing rows

### ISSUE THREAD

**Issue by vavrusa** [yandex/ClickHouse#164]
Hi,

I've encountered a strange issue with enums in index. So I have a table that looks somewhat like this:

```sql
CREATE TABLE IF NOT EXISTS logs_replica
(
	date Date,
	datetime DateTime,
	customerId UInt64,
        locationId UInt32,
        ... some other columns ...
	instance Enum8('other' = 0, 'external' = 1, 'internal' = 2)
)
ENGINE = ReplicatedMergeTree('...', '{replica}',
	date, locationId, (customerId, instance, datetime, locationId), 8192);
CREATE TABLE IF NOT EXISTS logs AS logs_replica
ENGINE = Distributed(cluster, default, logs_replica, rand());
```

Counting rows works fine, and so does GROUP BY

```
> SELECT COUNT(*)/1000000000 FROM logs
"divide(COUNT(), 1000000000)"
175.232771026
> SELECT COUNT(*)/1000000000 FROM logs GROUP BY instance
108.447320335
49.528222428
17.289928263
> SELECT COUNT(*)/1000000000 FROM logs WHERE instance IN 'other'
108.506876428
```

But it doesn't work when I use `instance = X` instead of `IN`, which I presumed should be equivalent. Notice the odd result:

```
> SELECT COUNT(*)/1000000000 FROM logs WHERE instance = 'other'
18.792497186
```

Same thing when I query local replica. With `enum = X` it select ~5x less rows than when using `IN` or `GROUP BY`. Is this a bug or am I doing something wrong?

**Comment by alexey-milovidov** [comment]
I will investigate this issue in few days.

**Comment by alexey-milovidov** [comment]
Confirmed. Enums does not work correctly for index.
```
:) CREATE TABLE test.enum_index (d Date DEFAULT '2000-01-01', x Enum8('a' = 1, 'b' = 2)) ENGINE = MergeTree(d, x, 1)

CREATE TABLE test.enum_index
(
    d Date DEFAULT '2000-01-01', 
    x Enum8('a' = 1, 'b' = 2)
) ENGINE = MergeTree(d, x, 1)

Ok.

0 rows in set. Elapsed: 0.017 sec. 

:) INSERT INTO test.enum_index (x) VALUES ('a'), ('a'), ('b'), ('b')

INSERT INTO test.enum_index (x) VALUES

Ok.

4 rows in set. Elapsed: 0.006 sec. 

:) SELECT * FROM test.enum_index WHERE x = 'a'

SELECT *
FROM test.enum_index 
WHERE x = 'a'

┌──────────d─┬─x─┐
│ 2000-01-01 │ b │
└────────────┴───┘
```

**Comment by vavrusa** [comment]
Looks like this was fixed in 1.1.54080, thanks gents! 🎉

**Comment by alexey-milovidov** [comment]
Yes, it is fixed. Thanks for the reminder!
